python/lsst/pipe/tasks/multiBand.py

```diff
@@ -180,11 +187,23 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
         idGenerator = self.config.idGenerator.apply(butlerQC.quantum.dataId)
         exposure = inputs.pop("exposure")
         assert not inputs, "runQuantum got more inputs than expected."
-        outputs = self.run(
-            exposure=exposure,
-            idFactory=idGenerator.make_table_id_factory(),
-            expId=idGenerator.catalog_id,
-        )
+        try:
+            outputs = self.run(
+                exposure=exposure,
+                idFactory=idGenerator.make_table_id_factory(),
+                expId=idGenerator.catalog_id,
+            )
+        except TooManyMaskedPixelsError as e:
+            error = AnnotatedPartialOutputsError(
+                e,
+                self,
+                exposure,
+                log=self.log,
+            )
+            result = Struct(outputSources=None, outputBackgrounds=None, outputExposure=None)
+            butlerQC.put(result, outputRefs)
```

Comment:
Since you're not actually writing anything, you can drop this line and the above it.